### DISCUSSION TITLE
群组白名单失效

### DISCUSSION THREAD

**Discussion by Eudiplozoon** [Tsuk1ko/CQ-picfinder-robot#210] [问题求助]
升级到2.29之后白名单出现失效的问题，问一下应该如何解决

**Comment by Tsuk1ko** [comment]
具体是哪些和白名单相关的功能失效

**Reply by Eudiplozoon** [reply]
应该是仅允许白名单群使用那里

**Reply by Tsuk1ko** [reply]
我测试了是正常的

**Reply by Eudiplozoon** [reply]
好吧 那我重新部署一遍试试，因为日志里好像也没看见相关报错。

**Reply by Miuzarte** [reply]
> 好吧 那我重新部署一遍试试，因为日志里好像也没看见相关报错。

是不是仅白名单的开关没开

**Reply by Eudiplozoon** [reply]
> > 好吧 那我重新部署一遍试试，因为日志里好像也没看见相关报错。
> 
> 是不是仅白名单的开关没开

不是 那个开关我是一直开着的，就这两周突然出现的问题。后来我搞了份默认的配置文件只更改白名单和仅允许白名单那两项，结果也是没用。

**Reply by Miuzarte** [reply]
那挺奇怪的  六月四那天怕腾讯ai加严我就打开了那个开关来着
'''json
      // 发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
      "deleteTime": 10,
      // 群内使用冷却时间（秒），每名用户独立，0 则无冷却，私聊无 cd
      "cd": 10,
      // 每名用户每日次数限制
      "limit": 100,
      // 群组白名单
      "whiteGroup": [],
      // 仅允许白名单群使用（与上面的私聊使用是独立的）
      "whiteOnly": true,
      // 白名单群组的群内使用冷却时间（秒），每名用户独立，0 则无冷却
      "whiteCd": 10,
      // 白名单群组内发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
      "whiteDeleteTime": 10,
      // r18 是否只发送地址不发送图片（如果开启，建议设置 `sendPximgProxys`）
      "r18OnlyUrl": {
        "private": false, // 私聊
        "group": true, // 群聊
        "temp": true // 群临时会话
      },
      // r18 图片只通过私聊发送（私聊不会被和谐，因此不会进行反和谐）
      "r18OnlyPrivate": true,
      // 允许临时会话发送 r18 结果，用法同 `bot.pmSearchResultTemp`
      "r18OnlyPrivateAllowTemp": false,
      // 只允许在白名单群组中发送 r18 setu
      "r18OnlyInWhite": false
'''

**Reply by Eudiplozoon** [reply]
> 那挺奇怪的 六月四那天怕腾讯ai加严我就打开了那个开关来着
> '''json
> // 发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
> "deleteTime": 10,
> // 群内使用冷却时间（秒），每名用户独立，0 则无冷却，私聊无 cd
> "cd": 10,
> // 每名用户每日次数限制
> "limit": 100,
> // 群组白名单
> "whiteGroup": [],
> // 仅允许白名单群使用（与上面的私聊使用是独立的）
> "whiteOnly": true,
> // 白名单群组的群内使用冷却时间（秒），每名用户独立，0 则无冷却
> "whiteCd": 10,
> // 白名单群组内发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
> "whiteDeleteTime": 10,
> // r18 是否只发送地址不发送图片（如果开启，建议设置 `sendPximgProxys`）
> "r18OnlyUrl": {
> "private": false, // 私聊
> "group": true, // 群聊
> "temp": true // 群临时会话
> },
> // r18 图片只通过私聊发送（私聊不会被和谐，因此不会进行反和谐）
> "r18OnlyPrivate": true,
> // 允许临时会话发送 r18 结果，用法同 `bot.pmSearchResultTemp`
> "r18OnlyPrivateAllowTemp": false,
> // 只允许在白名单群组中发送 r18 setu
> "r18OnlyInWhite": false
> '''

我在其他地方重新部署了一遍，白名单是正常的。也不知道原来那边是哪里出了问题QAQ 总之现在是能正常用了

**Reply by Eudiplozoon** [reply]
> 那挺奇怪的 六月四那天怕腾讯ai加严我就打开了那个开关来着
> '''json
> // 发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
> "deleteTime": 10,
> // 群内使用冷却时间（秒），每名用户独立，0 则无冷却，私聊无 cd
> "cd": 10,
> // 每名用户每日次数限制
> "limit": 100,
> // 群组白名单
> "whiteGroup": [],
> // 仅允许白名单群使用（与上面的私聊使用是独立的）
> "whiteOnly": true,
> // 白名单群组的群内使用冷却时间（秒），每名用户独立，0 则无冷却
> "whiteCd": 10,
> // 白名单群组内发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
> "whiteDeleteTime": 10,
> // r18 是否只发送地址不发送图片（如果开启，建议设置 `sendPximgProxys`）
> "r18OnlyUrl": {
> "private": false, // 私聊
> "group": true, // 群聊
> "temp": true // 群临时会话
> },
> // r18 图片只通过私聊发送（私聊不会被和谐，因此不会进行反和谐）
> "r18OnlyPrivate": true,
> // 允许临时会话发送 r18 结果，用法同 `bot.pmSearchResultTemp`
> "r18OnlyPrivateAllowTemp": false,
> // 只允许在白名单群组中发送 r18 setu
> "r18OnlyInWhite": false
> '''

我刚才在新部署的上又试了好几次，仅白名单好像还是没效果，，，

**Reply by Miuzarte** [reply]
> 我刚才在新部署的上又试了好几次，仅白名单好像还是没效果，，，

你复制我的配置上去试试
以及确认一下你是不是私聊的, 还有改完配置有没有保存到服务器, 保存好了--reload了吗
```json
      // 发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
      "deleteTime": 10,
      // 群内使用冷却时间（秒），每名用户独立，0 则无冷却，私聊无 cd
      "cd": 10,
      // 每名用户每日次数限制
      "limit": 100,
      // 群组白名单
      "whiteGroup": [],
      // 仅允许白名单群使用（与上面的私聊使用是独立的）
      "whiteOnly": true,
      // 白名单群组的群内使用冷却时间（秒），每名用户独立，0 则无冷却
      "whiteCd": 10,
      // 白名单群组内发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
      "whiteDeleteTime": 10,
      // r18 是否只发送地址不发送图片（如果开启，建议设置 `sendPximgProxys`）
      "r18OnlyUrl": {
        "private": false, // 私聊
        "group": true, // 群聊
        "temp": true // 群临时会话
      },
      // r18 图片只通过私聊发送（私聊不会被和谐，因此不会进行反和谐）
      "r18OnlyPrivate": true,
      // 允许临时会话发送 r18 结果，用法同 `bot.pmSearchResultTemp`
      "r18OnlyPrivateAllowTemp": false,
      // 只允许在白名单群组中发送 r18 setu
      "r18OnlyInWhite": false
```

**Reply by Eudiplozoon** [reply]
> > 我刚才在新部署的上又试了好几次，仅白名单好像还是没效果，，，
> 
> 你复制我的配置上去试试
> 以及确认一下你是不是私聊的, 还有改完配置有没有保存到服务器, 保存好了--reload了吗
> 
> ```json
>       // 发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
>       "deleteTime": 10,
>       // 群内使用冷却时间（秒），每名用户独立，0 则无冷却，私聊无 cd
>       "cd": 10,
>       // 每名用户每日次数限制
>       "limit": 100,
>       // 群组白名单
>       "whiteGroup": [],
>       // 仅允许白名单群使用（与上面的私聊使用是独立的）
>       "whiteOnly": true,
>       // 白名单群组的群内使用冷却时间（秒），每名用户独立，0 则无冷却
>       "whiteCd": 10,
>       // 白名单群组内发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
>       "whiteDeleteTime": 10,
>       // r18 是否只发送地址不发送图片（如果开启，建议设置 `sendPximgProxys`）
>       "r18OnlyUrl": {
>         "private": false, // 私聊
>         "group": true, // 群聊
>         "temp": true // 群临时会话
>       },
>       // r18 图片只通过私聊发送（私聊不会被和谐，因此不会进行反和谐）
>       "r18OnlyPrivate": true,
>       // 允许临时会话发送 r18 结果，用法同 `bot.pmSearchResultTemp`
>       "r18OnlyPrivateAllowTemp": false,
>       // 只允许在白名单群组中发送 r18 setu
>       "r18OnlyInWhite": false
> ```

我终于知道问题了，，，是我理解错了不好意思，这个仅白名单是针对setu功能的啊，我一直以为是所有bot相关功能都仅对白名单生效。
现在只能通过忽略群/人来实现全功能的白名单，不知道之后的版本能不能在配置文件里加个全功能白名单呢QAQ

**Reply by Tsuk1ko** [reply]
可以
